bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/views/markers/MarkerEntry.java

```diff
@@ -219,7 +221,8 @@ CollationKey getCollationKey(String attribute, String defaultValue) {
 		if (attributeValue.isEmpty()) {
 			return MarkerSupportInternalUtilities.EMPTY_COLLATION_KEY;
 		}
-		CollationKey key = Collator.getInstance().getCollationKey(attributeValue);
+		CollationKey key = collationCache.computeIfAbsent(attributeValue,
+				k -> Collator.getInstance().getCollationKey(attributeValue));
```

Comment:
I haven't tried that in the debugger yet, but what kind of values are cached here, and why is this cache **per marker**? Shouldn't that be cache per view, or static one?